src/ir/assigments.rs

```diff
@@ -28,6 +27,19 @@ pub struct AssigmentGenerator<F, TraceArgs> {
     ir_id: UUID,
 }
 
+impl<F: Clone, TraceArgs> Clone for AssigmentGenerator<F, TraceArgs> {
+    fn clone(&self) -> Self {
+        Self {
+            columns: self.columns.clone(),
+            placement: self.placement.clone(),
+            selector: self.selector.clone(),
+            trace_gen: self.trace_gen.clone(),
+            num_rows: self.num_rows,
+            ir_id: self.ir_id,
+        }
+    }
+}
+
```

Comment:
How is it different from the default implementation? `usize` and `UUID` should be both copied rather than cloned in this implementation, but that isn't much different. I probably am missing something?

Reply:
The derived implementation requires TraceArgs to be Clone, when it is not necessary.